core/src/main/java/io/smallrye/openapi/runtime/io/JacksonJsonIO.java

```diff
@@ -334,7 +339,7 @@ public Object fromJson(JsonNode value) {
     @SuppressWarnings("unchecked")
     @Override
     public <T> T fromJson(JsonNode object, Class<T> desiredType) {
-        if (desiredType == String.class) {
+        if (desiredType == String.class && object.isValueNode()) {
```

Comment:
I added this so that `{}` and `[]` are not coerced to empty strings whereas any value type will be. This matches the Jakarta JSON reader (which I don't think anyone uses yet), but also seems more intuitive. wdyt?

Reply:
Yeah, that makes sense to me.